metaflow/decorators.py

```diff
@@ -99,7 +100,8 @@ def _parse_decorator_spec(cls, deco_spec):
             return cls()
         else:
             name, attrspec = top
-            attrs = dict(a.split('=') for a in attrspec.split(','))
+            attrs = dict(a.split('=') for a in 
+                re.split(''',(?=(?:[^'"]|'[^']*'|"[^"]*")*$)''', attrspec))
```

Comment:
Another suggestion for this regexp would be something of this nature:
`(?=[\s\w]+=)`. It basically matches commas that are "top-level" in the sense that they directly precede another attribute of the type name=value. This would allow you to parse exactly the way you would specify in a step.As an example, we have in the tutorial:
`libraries={'editdistance': '0.5.3', 'pandas' : '0.24.2'}, python=3.8.1` which would be parseable by this method.

Further, I would possibly add a few strips to allow for spaces again to make it more legible for the user:
`attrs = dict(map(lambda x: x.strip(), a.split('=')) for a in re.split(''',(?=[\s\w]+=)''', attrspec.strip('"\'')))`

Reply:
Done.